interface/modules/custom_modules/oe-module-dashboard-context/public/admin.php

```diff
@@ -0,0 +1,1105 @@
+                        const isActive = widgetConfig[widgetId] !== false;
+                        $widgetGrid.append(
+                            $('<div>', {class: 'widget-item ' + (isActive ? 'active' : '')}).append(
+                                $('<div>', {class: 'form-check mb-0'}).append(
+                                    $('<input>', {
+                                        type: 'checkbox',
+                                        class: 'form-check-input widget-toggle',
+                                        id: 'widget-' + widgetId,
+                                        'data-widget': widgetId,
+                                        checked: isActive
+                                    }),
+                                    $('<label>', {
+                                        class: 'form-check-label',
+                                        for: 'widget-' + widgetId
+                                    }).text(widgetLabel)
+                                )
+                            )
+                        );
+                    });
+
+                    var $dialog = $('<div>', {id: 'contextDialog', class: 'dialog-overlay'}).append(
+                        $('<div>', {class: 'dialog-content', style: 'width:90%;max-width:700px;'}).append(
+                            $('<div>', {class: 'dialog-header bg-secondary text-light'}).append(
+                                $('<h5>', {class: 'm-0'}).text(isEdit ? self.xl.editContext : self.xl.createContext),
+                                $('<button>', {type: 'button', class: 'btn btn-sm btn-light dialog-close'}).html('&times;')
+                            ),
+                            $('<div>', {class: 'dialog-body'}).append(
+                                $('<input>', {type: 'hidden', id: 'contextId', value: isEdit ? contextData.id : ''}),
+                                $('<div>', {class: 'form-group'}).append(
+                                    $('<label>', {class: 'font-weight-bold'}).text(self.xl.contextName + ' *'),
+                                    $('<input>', {
+                                        type: 'text',
+                                        class: 'form-control',
+                                        id: 'contextName',
+                                        value: isEdit ? contextData.context_name : ''
+                                    })
+                                ),
+                                $('<div>', {class: 'form-group'}).append(
+                                    $('<label>', {class: 'font-weight-bold'}).text(self.xl.contextKey),
+                                    $('<input>', {
+                                        type: 'text',
+                                        class: 'form-control',
+                                        id: 'contextKey',
+                                        value: isEdit ? contextData.context_key : '',
```

Comment:
do these need to be wrapped with text() to prevent xss (like you do in other places in this script)?
if so, rec doing that also in other placed in this scrip.